syntax = "proto2";

message Summary {

  message OnlineContentApps {
    optional float processes__avg = 1;
    optional float hosts__avg = 2;
  }

  optional OnlineContentApps online_content_apps = 1;


  message OnlineWorkers {
    optional float processes__avg = 1;
    optional float hosts__avg = 2;
  }

  optional OnlineWorkers online_workers = 2;


  message XYComponent {
    required string name = 1;
    required string version = 2;
    required uint32 count = 3;
  }

  repeated XYComponent xy_component = 3;


  message XYZComponent {
    required string name = 1;
    required string version = 2;
    required uint32 count = 3;
  }

  repeated XYZComponent xyz_component = 4;

  message NumberCount {
    required uint32 number = 1;
    required uint32 count = 2;
  }

  message AgeCount {
    required uint32 age = 1;
    required uint32 count = 2;
  }

  repeated AgeCount age_count = 5;

  message PostgresqlVersion {
    required uint32 version = 1;
    required uint32 count = 2;
  }

  repeated PostgresqlVersion postgresql_version = 6;

  message RbacStats {
    repeated NumberCount users = 1;
    repeated NumberCount groups = 2;
    repeated NumberCount domains = 3;
    repeated NumberCount custom_access_policies = 4;
    repeated NumberCount custom_roles = 5;
  }

  optional RbacStats rbac_stats = 7;

}
